daemon/buildinfo.c

```diff
@@ -367,41 +367,104 @@ void print_build_info_json(void) {
     printf("}\n");
 };
 
-//return a list of enabled features for use in analytics
-//find a way to have proper |
+#define add_to_bi(buffer, str)       \
+    { if(first) {                    \
+        buffer_strcat (b, str);      \
+        first = 0;                   \
+    } else                           \
+        buffer_strcat (b, "|" str); }
+
 void analytics_build_info(BUFFER *b) {
-    if(FEAT_DBENGINE)        buffer_strcat (b, "dbengine");
-    if(FEAT_NATIVE_HTTPS)    buffer_strcat (b, "|Native HTTPS");
-    if(FEAT_CLOUD)           buffer_strcat (b, "|Netdata Cloud");
-    if(FEAT_CLOUD)           buffer_strcat (b, "|ACLK Next Generation");
-    if(NEW_CLOUD_PROTO)      buffer_strcat (b, "|New Cloud Protocol Support");
-    if(FEAT_TLS_HOST_VERIFY) buffer_strcat (b, "|TLS Host Verification");
-    if(FEAT_ML)              buffer_strcat (b, "|Machine Learning");
-    if(FEAT_STREAM_COMPRESSION) buffer_strcat (b, "|Stream Compression");
-
-    if(FEAT_PROTOBUF)        buffer_strcat (b, "|protobuf");
-    if(FEAT_JEMALLOC)        buffer_strcat (b, "|jemalloc");
-    if(FEAT_JSONC)           buffer_strcat (b, "|JSON-C");
-    if(FEAT_LIBCAP)          buffer_strcat (b, "|libcap");
-    if(FEAT_CRYPTO)          buffer_strcat (b, "|libcrypto");
-    if(FEAT_LIBM)            buffer_strcat (b, "|libm");
-
-    if(FEAT_TCMALLOC)       buffer_strcat(b, "|tcalloc");
-    if(FEAT_ZLIB)           buffer_strcat(b, "|zlib");
-
-    if(FEAT_APPS_PLUGIN)    buffer_strcat(b, "|apps");
-    if(FEAT_CGROUP_NET)     buffer_strcat(b, "|cgroup Network Tracking");
-    if(FEAT_CUPS)           buffer_strcat(b, "|CUPS");
-    if(FEAT_EBPF)           buffer_strcat(b, "|EBPF");
-    if(FEAT_IPMI)           buffer_strcat(b, "|IPMI");
-    if(FEAT_NFACCT)         buffer_strcat(b, "|NFACCT");
-    if(FEAT_PERF)           buffer_strcat(b, "|perf");
-    if(FEAT_SLABINFO)       buffer_strcat(b, "|slabinfo");
-    if(FEAT_XEN)            buffer_strcat(b, "|Xen");
-    if(FEAT_XEN_VBD_ERROR)  buffer_strcat(b, "|Xen VBD Error Tracking");
-
-    if(FEAT_KINESIS)        buffer_strcat(b, "|AWS Kinesis");
-    if(FEAT_PUBSUB)         buffer_strcat(b, "|GCP PubSub");
-    if(FEAT_MONGO)          buffer_strcat(b, "|MongoDB");
-    if(FEAT_REMOTE_WRITE)   buffer_strcat(b, "|Prometheus Remote Write");
+    int first = 1;
+#ifdef ENABLE_DBENGINE
```

Comment:
Hello @underhood these tests you are using here were used to define variables like `FEAT_DBENGINE`, is there a special motive for you discard these definitions and reuse the same tests here?

Reply:
so it is skipped completely by preprocessor